client/src/main/java/org/eclipse/hono/client/ServerErrorException.java

```diff
@@ -21,6 +21,8 @@
 
     private static final long serialVersionUID = 1L;
 
+    private String clientFacingMessage;
```

Comment:
this looks like it would be helpful in both ClientErrorException and ServerErrorException. Thus, IMHO we should add this to ServiceInvocationException.

Reply:
The motivation to put it only in the `ServerErrorException` class was that the standard error message of a `ClientErrorException` is already a client-facing one, i.e. it is used in the HTTP response for example:
https://github.com/eclipse/hono/blob/0e440efe3e6fa1d28f5bd4543137fbdda3b609ea/adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java#L787-L789

I see a `ServerErrorException` as more of an internal exception, where it makes sense to add a separate error message that abstracts from the internals.

Reply:
I see